core/math/aabb.cpp

```diff
@@ -135,27 +134,30 @@ bool AABB::intersects_ray(const Vector3 &p_from, const Vector3 &p_dir, Vector3 *
 				return false;
 			}
 		} else { // ray not parallel to planes in this direction
-			c1[i] = (position[i] - p_from[i]) / p_dir[i];
-			c2[i] = (end[i] - p_from[i]) / p_dir[i];
+			real_t t1 = (position[i] - p_from[i]) / p_dir[i];
+			real_t t2 = (end[i] - p_from[i]) / p_dir[i];
 
-			if (c1[i] > c2[i]) {
-				SWAP(c1, c2);
+			if (t1 > t2) {
+				SWAP(t1, t2);
 			}
-			if (c1[i] > near) {
-				near = c1[i];
+
+			if (t1 > tmin) {
```

Comment:
If `tmin` is initialized to 0.0, if the start point is on the exact edge, and it moves towards the box, will this `if` fail, and thus not store the axis for the returned normal? :thinking: 

Could `>=` be better?

Reply:
no because first case wins and you want to avoid extra ops where possible this code should be VERY performant. As rays vs boxes is pretty common. The nasty case of being on the corner of a box which happens more often than you realise is degenerate and your best to return the first result asap